cmd/telegraf/telegraf.go

```diff
@@ -271,7 +271,7 @@ func reloadLoop(stop chan struct{}, s service.Service) {
 		log.Printf("I! Tags enabled: %s", c.ListTags())
 
 		if *fPidfile != "" {
-			f, err := os.Create(*fPidfile)
+			f, err := os.OpenFile(*fPidfile, os.O_EXCL|os.O_CREATE|os.O_WRONLY, 0666)
```

Comment:
> The alternative is checking if the pid in the pidfile refers to a live process, and if not, reclaiming the file for ourselves.

The problem is that with things like PID reuse, that can't be relied upon. It might not be a telegraf process sitting on that PID. And checking that the pid exists is a little difficult to do in a good cross-platform way. The best I know if is to send a signal 0 to the PID, and check response code. But even that can be misleading as you can't signal processes that you don't have perms on. So it gets messy really fast.